### PR TITLE
chore(source-s3): Update CDK to v5 (old)

### PR DESCRIPTION
**Pull request by aaronsteers**
Replaces:

- https://github.com/airbytehq/airbyte/pull/45186

This bumps the CDK version so we can take advantage of performance boosts in latest CDK.

I spun this out from @artem1205's PR above in order to resolve some CAT failures.

I've imported `integration_tests/test_acceptance.py` from the S3 Polars PR. And I added another second test, which asserts that the invalid config json fails as expected.

### CODE REVIEW

File: airbyte-integrations/connectors/source-s3/integration_tests/spec.json [hunk 1]
```diff
@@ -338,6 +340,21 @@
...
+                    "filetype": {
+                      "title": "Filetype",
+                      "default": "excel",
+                      "const": "excel",
+                      "type": "string"
+                    }
+                  },
+                  "required": [
+                    "filetype"
+                  ]
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-s3/integration_tests/spec.json:357]:
```suggestion
                  "required": ["filetype"]
```

File: airbyte-integrations/connectors/source-s3/integration_tests/test_acceptance.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+"""Run a subset of acceptance tests in PyTest.
+
+This test file provides a means to easily step through a subset of acceptance tests, debugging
+and verifying the behavior of the source-s3 connector. The tests are redundant with CAT, but
+are much faster to run, and easier to run and easier to debug.
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-s3/integration_tests/test_acceptance.py:5]:
```suggestion
are much faster to run, and easier to debug.
```

File: airbyte-integrations/connectors/source-s3/integration_tests/cloud_spec.json [hunk 3]
```diff
@@ -385,10 +402,12 @@
       },
       "endpoint": {
         "title": "Endpoint",
-        "description": "Endpoint to an S3 compatible service. Leave empty to use AWS. The custom endpoint must be secure, but the 'https' prefix is not required.",
+        "description": "Endpoint to an S3 compatible service. Leave empty to use AWS.",
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-s3/integration_tests/cloud_spec.json:405]:
```suggestion
        "description": "Endpoint to an S3 compatible service. Leave empty to use AWS. The custom endpoint must be secure, but the 'https' prefix is not required.",
```

File: airbyte-integrations/connectors/source-s3/integration_tests/cloud_spec.json [hunk 4]
```diff
@@ -385,10 +402,12 @@
...
         "default": "",
-        "examples": ["my-s3-endpoint.com", "https://my-s3-endpoint.com"],
-        "pattern": "^(?!http://).*$",
+        "examples": [
+          "my-s3-endpoint.com",
+          "https://my-s3-endpoint.com"
+        ],
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-s3/integration_tests/cloud_spec.json:410]:
```suggestion
        ],
        "pattern": "^(?!http://).*$",
```

Comment (aaronsteers) [inline airbyte-integrations/connectors/source-s3/integration_tests/cloud_spec.json:410]:
```suggestion
        "examples": ["my-s3-endpoint.com", "https://my-s3-endpoint.com"],
        "pattern": "^(?!http://).*$",
```

File: airbyte-integrations/connectors/source-s3/source_s3/run.py [hunk 5]
```diff
@@ -27,17 +28,21 @@ def get_source(args: List[str]):
...
-                    ),
-                ),
-            ).json()
+            orjson.dumps(
+                AirbyteMessageSerializer.dump(
+                    AirbyteMessage(
+                        type=Type.TRACE,
+                        trace=AirbyteTraceMessage(
+                            type=TraceType.ERROR,
+                            emitted_at=time.time_ns() // 1_000_000,
```
Comment (girarda) [inline airbyte-integrations/connectors/source-s3/source_s3/run.py:37]: good catch!

**Review by girarda — APPROVED** [review]

**Comment by natikgadzhi** [comment]
@aaronsteers is this abandoned in favor of the other one? Should we close this?

**Comment by aaronsteers** [comment]
@natikgadzhi Re:

> @aaronsteers is this abandoned in favor of the other one? Should we close this?

No, this one was spun up because of headaches with breaking changes unrelated to the larger refactorings here:

- #44194 

I've bumped this both to draft status, fwiw. The order of operations once I'm off of oncall rotation is:

1. Get the CDK bumped.
2. Fix the concurrent CDK bug, which otherwise prevents file-based CDK to use concurrency in specific instances.
3. Larger refactoring.

Since we're expecting a decent perf boost from 1 and 2, number 3 is not "unstoppable" on its own.

Some of this context also shared in Slack. And I've created a new channel [#proj-s3-performance-boosts](https://airbytehq-team.slack.com/archives/C07QM5QTCA2) for focused conversations on this initiative.

**Comment by aaronsteers** [comment]
@natikgadzhi Re:

> @aaronsteers is this abandoned in favor of the other one? Should we close this?

Yes. There's still some code here I eventually plan to recycle, but the new effort is here:

- #46298

**Pull request closed without merging**